util/vkgcMetroHash.h

```diff
@@ -86,6 +86,13 @@ inline bool operator!=(const Hash &lhs, const Hash &rhs) {
 namespace std {
 template <> struct hash<MetroHash::Hash> {
   // Returns `hash` compacted to `size_t`. Returns zero for value-initialized hashes.
-  size_t operator()(const MetroHash::Hash &hash) const { return static_cast<size_t>(MetroHash::compact64(&hash)); }
+  size_t operator()(const MetroHash::Hash &hash) const {
+    if constexpr (sizeof(size_t) == 8) {
```

Comment:
constexpr is probably not required - but it makes it more explicit what is happening.